http/config.go

```diff
@@ -0,0 +1,144 @@
+package http
+
+import (
+	"net/http"
+	"net/url"
+	"sync"
+
+	cors "github.com/rs/cors"
+)
+
+const (
+	ACAOrigin      = "Access-Control-Allow-Origin"
+	ACAMethods     = "Access-Control-Allow-Methods"
+	ACACredentials = "Access-Control-Allow-Credentials"
+)
+
+type ServerConfig struct {
+	// APIPath is the prefix of all request paths.
+	// Example: host:port/api/v0/add. Here the APIPath is /api/v0
+	APIPath string
+
+	// Headers is an optional map of headers that is written out.
+	Headers map[string][]string
+
+	// corsOpts is a set of options for CORS headers.
+	corsOpts *cors.Options
+
+	// corsOptsRWMutex is a RWMutex for read/write CORSOpts
+	corsOptsRWMutex sync.RWMutex
```

Comment:
I like this pattern that @Kubuxu pointed out recently: https://play.golang.org/p/ddVfHDYu06Y

Reply:
Yes, that's neat. I don't take authorship for this part though, I only moved it from handler.go :)